cg/meta/orders/external_data.py

```diff
@@ -0,0 +1,186 @@
+        sbatch_number: int = slurm_api.submit_sbatch(
+            sbatch_content=sbatch_content, sbatch_path=sbatch_path
+        )
+        return sbatch_number
+
+    def download_ticket(self, ticket_id: int, dry_run: bool) -> None:
+        cases: List[models.Family] = self.status_db.get_cases_from_ticket(ticket_id=ticket_id).all()
+        cust_id = cases[0].customer.internal_id
+        for case in cases:
+            links = case.links
+            for link in links:
+                lims_sample_id = link.sample.internal_id
+                cust_sample_id = link.sample.name
+                sbatch_number = self.download_sample(
+                    cust_id=cust_id,
+                    ticket_id=ticket_id,
+                    cust_sample_id=cust_sample_id,
+                    lims_sample_id=lims_sample_id,
+                    dry_run=dry_run,
+                )
+                LOG.info(
+                    "Downloading sample %s from caesar by submitting sbatch job %s"
+                    % (lims_sample_id, sbatch_number)
+                )
+
+    def get_all_fastq(self, sample_folder: str) -> List[str]:
+        all_fastqs = []
+        for leaf in os.listdir(sample_folder):
+            abs_path = sample_folder + leaf
+            LOG.info("Found file %s inside folder %s" % (abs_path, sample_folder))
+            if abs_path.endswith("fastq.gz"):
+                all_fastqs.append(abs_path)
+        return all_fastqs
+
+    def get_all_paths(self, lims_sample_id: str, cust_id: str) -> List[str]:
+        fastq_folder: str = self.create_destination_path(
+            cust_id=cust_id, lims_sample_id=lims_sample_id, raw_path=self.hasta_path
+        )
+        all_fastq_in_folder: List[str] = self.get_all_fastq(sample_folder=fastq_folder)
+        return all_fastq_in_folder
+
+    def create_data_dict(self, name: str, file_list: List[dict]) -> dict:
+        timestamp = dt.datetime.now()
+        hk_dict = {
+            "name": name,
+            "created_at": timestamp,
+            "expires_at": None,
+            "files": file_list,
+        }
+        return hk_dict
+
+    def create_file_list(self, lims_id: str, cust_id: str) -> List[dict]:
+        file_list: List[dict] = []
+        for file in self.get_all_paths(lims_sample_id=lims_id, cust_id=cust_id):
+            file_list.append({"path": file, "tags": ["fastq"], "archive": False})
+        return file_list
+
+    def create_hk_bundle(
+        self, bundle_name: str, dry_run: bool, data_dict: dict
+    ) -> Tuple[Bundle, Version]:
+        if not dry_run:
+            bundle_result: Tuple[Bundle, Version] = self.housekeeper_api.add_bundle(
+                bundle_data=data_dict
+            )
+            return bundle_result
+        else:
+            LOG.info(
+                "Would have created bundle %s in housekeeper, but this is dry-run", bundle_name
+            )
+
+    def get_all_cases_from_ticket(self, ticket_id: int):
```

Comment:
type hint

Reply:
Now I know why I forgot type hint, it was because this was the fulfunktion that you were going to help me replace today😇